enumerables.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    conv = if is_a? Range
+             to_a
+           else
+             self
+           end
+    i = 0
+    while i < conv.length
+      yield conv[i]
+      i += 1
+    end
+    conv
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    conv = if is_a? Range
+             to_a
+           else
+             self
+           end
+    i = 0
+    while i < conv.length
+      yield conv[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    selection = []
+    my_each { |item| selection << item if yield(item) }
+    selection
+  end
+
+  def my_all?(arg = nil)
+    check = true
+    if arg
+      my_each { |i| check = false unless arg === i } # rubocop:disable Style/CaseEquality
+    else
+      my_each { |element| check = false unless yield(element) }
+    end
+    check
+  end
```

Comment:
* [ ] should have working `my_all?`  identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_all? 

- [ ] when no block or argument is given returns true only when none of the collection members are false or nil, if the mentioned condition is not met it, should return false

```ruby     
[nil, false, true, []].my_all? == [nil, false, true, []].all? => false
```
- Yours is giving LocalJumpError (always expecting a block)
![image](https://user-images.githubusercontent.com/52209067/80495431-2376b980-8925-11ea-8737-804762ae4375.png)



- Real all?
![image](https://user-images.githubusercontent.com/52209067/77688266-67148700-6f65-11ea-82e5-cbef16444817.png)